util/chplenv/chpl_aux_filesys.py

```diff
@@ -20,6 +21,24 @@ def get():
             stderr.write("Warning: Can't find your Java installation\n")
         if not found_hdfs:
             stderr.write("Warning: Can't find your Hadoop installation\n")
+
+    elif aux_fs == 'hdfs3':
+        def fetchInfo(env, val, err):
+            stripstr = (env.find('INCLUDE') > 0 and '-I' or '-L')
+            files = map(lambda z: z.lstrip(stripstr), os.environ.get(env, '').split())
+            res = 0
+            for f in files:
+               res += sum(map(lambda x: 1, filter(lambda z: z.find(val) > 0 and 1 or 0, glob("%s/*" % (f,)))))
+
+            if res < 1:
+                stderr.write(err)
+                return False
+
+            return True
+
+        fetchInfo('CHPL_AUXIO_INCLUDE', 'hdfs', "Warning: Can't find your HDFS3 header file installation\n")
+        fetchInfo('CHPL_AUXIO_LIBS', 'hdfs3', "Warning: Can't find your HDFS3 static library installation\n")
```

Comment:
since libhdfs3 has a library name different than the apache hadoop library name, so this was an attempt to make the search for header files and library files a little more robust. the 'sum(map(lambda' business is cycling through all the files in the directory provided for -L or -I,  testing if each file name have an 'hfs' or 'hdfs3'. since it's a map a list is being made - a list of 1 or 0 is generated (1 for string match) the sum collapses values. at the other side of that loop res will be either greater than 0 or less than 1.